### PR TITLE
Run tests and linters with Github Actions CI

### PR DESCRIPTION
**Pull request by Fenthick**
Use Super-Linter to run multiple static analysis tools for multiple files.
Use golangci-lint for the Go code
Fix code to comply to linters

Note that "test" job could use Actions caching functionality to cache downloaded
Go modules and Go build cache. This improvement could be configured in
the future.

Issue: KNTK-182

### CODE REVIEW

**Comment by Fenthick** [comment]
Here you can see the output of the workflow: https://github.com/Fenthick/community_sdk_golang/actions/runs/599250103

File: kentikapi/internal/api_payloads/device.go [hunk 1]
```diff
@@ -336,6 +336,7 @@ func payloadToDeviceLabel(p deviceLabelPayload) (models.DeviceLabel, error) {
 	}, nil
 }
 
+// nolint: deadcode, unused
```
Comment (Fenthick) [inline kentikapi/internal/api_payloads/device.go:339]: @mateuszmidor Is it going to be used in the near future? Otherwise I'll remove the function

Comment (mateuszmidor) [reply kentikapi/internal/api_payloads/device.go:339]: You can remove it.

File: kentikapi/internal/api_resources/devices_test.go [hunk 2]
```diff
@@ -112,7 +113,7 @@ func TestCreateDeviceRouter(t *testing.T) {
 	models.SetOptional(&router.DeviceBGPPassword, "bgp-optional-password")
 	models.SetOptional(&router.SiteID, 8483)
 	models.SetOptional(&router.DeviceBGPFlowSpec, true)
-	device, err := devicesAPI.Create(nil, *router)
```
Comment (mateuszmidor) [inline kentikapi/internal/api_resources/devices_test.go:115]: StubTransport doesnt use the context

File: kentikapi/internal/api_payloads/device.go (hunk 1 above)
Comment (Fenthick) [reply kentikapi/internal/api_payloads/device.go]: Done

**Review by mateuszmidor — APPROVED** [review]

File: kentikapi/internal/api_resources/devices_test.go (hunk 2 above)
Comment (Fenthick) [reply kentikapi/internal/api_resources/devices_test.go:115]: Ok, but I did this change to suppress the linter output. And I guess this check is based on the Context package documentation (https://pkg.go.dev/context@go1.16#pkg-overview):
> Do not pass a nil Context, even if a function permits it. Pass context.TODO if you are unsure about which Context to use. 

Do you suggest to disable this check globally / locally?
In my opinion the best option here is just to pass `Background()` here and move on.

Comment (mateuszmidor) [reply kentikapi/internal/api_resources/devices_test.go:115]: agree, no point in fighting the tool here

**Review by lukaa12 — APPROVED** [review]

**Review by TedatKentik — APPROVED** [review]
updating tests and linters

**Pull request merged**